homeassistant/components/denonavr/media_player.py

```diff
@@ -131,6 +137,79 @@ async def async_setup_entry(
     async_add_entities(entities, update_before_add=True)
 
 
+def async_log_errors(
+    func: Callable[Concatenate[_DenonDeviceT, _P], Awaitable[_R]],
+) -> Callable[Concatenate[_DenonDeviceT, _P], Coroutine[Any, Any, _R | None]]:
```

Comment:
The change here looks bigger than it really is.
I have mostly just moved `async_log_errors` out of the `DenonDevice` to be able to type it correctly.
It was also missing a return value in case of failure in the previous implementation, now it returns `None`.